pkg/operator/resources/namespaced/controller.go

```diff
@@ -175,14 +171,14 @@ func createControllerServiceAccount() *corev1.ServiceAccount {
 
 func createControllerDeployment(controllerImage, importerImage, clonerImage, uploadServerImage, verbosity, pullPolicy string, imagePullSecrets []corev1.LocalObjectReference, priorityClassName string, infraNodePlacement *sdkapi.NodePlacement, replicas int32) *appsv1.Deployment {
 	defaultMode := corev1.ConfigMapVolumeSourceDefaultMode
-	deployment := utils.CreateDeployment(controllerResourceName, "app", "containerized-data-importer", common.ControllerServiceAccountName, imagePullSecrets, int32(1), infraNodePlacement)
+	deployment := utils.CreateDeployment(common.CDIControllerResourceName, common.CDIComponentLabel, common.CDIControllerResourceName, common.ControllerServiceAccountName, imagePullSecrets, int32(1), infraNodePlacement)
```

Comment:
So what happens is that this is breaking the upgrade path (which is exercised by CI):
```
2024-01-23T11:52:20Z	ERROR	cdi-operator		{"Request.Namespace": "", "Request.Name": "cdi", "error": "Deployment.apps \"cdi-deployment\" is invalid: spec.selector: Invalid value: v1.LabelSelector{MatchLabels:map[string]string{\"cdi.kubevirt.io\":\"cdi-deployment\"}, MatchExpressions:[]v1.LabelSelectorRequirement(nil)}: field is immutable"}
```

This PR changes the immutable deployment field `spec.selector` and thus the operator fails to reconcile the (already existing) cdi-deployment resource that is carried over from the previous version.

My suggestion is to stick to the existing selector we have

Reply:
@yaroslavborbat you can test upgrade locally by setting

```bash
export UPGRADE_FROM=v1.58.0

make cluster-sync
```

Reply:
Thank you all for your help. I fixed the problem and now all the tests have been successful